File: SW Expert Academy/D2/01989. 초심자의 회문 검사/초심자의 회문 검사.cpp
#include <iostream>
using namespace std;

int main(){
	int T; cin>>T;
	for(int tc=0; tc<T; tc++){
		string s; cin>>s;
		
		printf("#%d ", tc+1);
		
		// 투 포인터 형식
		int left=0, right=s.length()-1;
		bool flag = true;
		for(; left<=right; left++, right--){
			if(s[left] != s[right]){
				flag = false;
				printf("0\n");
				break;
			}
		}
		
		if(flag) printf("1\n");
	}
}

File: SW Expert Academy/D2/01989. 초심자의 회문 검사/초심자의 회문 검사.py
for tc in range(int(input())):
    s = input()

    print("#%d"%(tc+1), end=' ')

    left,right = 0,len(s)-1
    flag = True
    while left <= right:
        if s[left] != s[right]:
            flag = False
            print(0)
            break
        else: left,right = left+1,right-1

    if flag: print(1)
